math/cos_test.go

```diff
@@ -5,7 +5,7 @@ import (
 	"testing"
 )
 
-const epsilon = 0.001
+const epsilonCos = 0.001
```

Comment:
Change the package name to `math_test` in this file as well.

Reply:
> Change the package name to `math_test` in this file as well.

create new package math_test?

Reply:
Are you not aware of the `_test` feature of Go tests?

The following is an explanation for why there is a distinction - ignore it if you already know the difference between `package a` and `package a_test`. 

There are a few things to consider when writing tests (at least for me). First thing to consider is whether the test is an integration test or unit test. Unit tests usually live inside the package they were developed (in this case `math`). Integration tests however are a bit more technical, where you have the option to choose the same package or a different package (Go allows you to have another package in the same directory with `_test` suffix). Now the tests that you have are integration tests because they are being performed on exported function (think public methods in Java).

In practice, your methods are simple so it doesn't matter what kind of tests you do but I have been thinking recently to change a few things in the repo (whenever I get some time). But having a separate package (in the same folder with `_test` suffix) makes it easy to test API behaviour.


Consider as an example:


```Go
package something

func someInternalFunction(input int) error { ... }

func PublicApiFunction() error {
    input := <calculated_input>
    return someInternalFunction(input)
}

```

Now to test the `someInternalFunction`, you would need to write test inside the `something` package because its only available to the members of this package - the tests are members of this package since we have `package something` in the beginning of our test files. Note that you can also test the `PubliApiFunction` directly inside this package as well however, from a Public API perspective, we don't care about what the internal behaviour of the `PublicApiFunction` is and so there might be situations where the behaviour of our `PublicApiFunction` is different outside the package. This is where integration tests come in, and you can use the `package something_test` declaration in the beginning of a file in the same directory as you package and import the `something` package how you would in your code outside the package. This kind of test ensures that there is no leak in our Public API logic and would behave as we expect would it to behave.

Reply:
> 

Thank you very much! Done